rclpy/test/test_time.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from rclpy.clock import ClockType
+from rclpy.duration import Duration
+from rclpy.time import Time
+
+from test_msgs.msg import Builtins
+
+
+class TestTime(unittest.TestCase):
+
+    def test_time_construction(self):
+        time = Time()
+        assert time.nanoseconds == 0
+
+        time = Time(seconds=1, nanoseconds=5e8, clock_type=ClockType.SYSTEM_TIME)
+        assert time.nanoseconds == 1500000000
+        assert time.clock_type == ClockType.SYSTEM_TIME
+
+        with self.assertRaises(ValueError):
+            time = Time(seconds=-1)
+        with self.assertRaises(ValueError):
+            time = Time(nanoseconds=-1)
+        with self.assertRaises(TypeError):
+            time = Time(clock_type='SYSTEM_TIME')
+
+    def test_time_operators(self):
+        time1 = Time(nanoseconds=1, clock_type=ClockType.STEADY_TIME)
+        time2 = time1
```

Comment:
What is this supposed to test? Assignment makes `time2` and `time1` be references to the same instance of `Time`.

```
>>> t1 = Time(nanoseconds=1)
>>> t2 = t1
>>> id(t2)
140468253530040
>>> id(t1)
140468253530040
>>> id(t2) == id(t1)
True
````

Reply:
ah, my intent was more to check how the copies interacted. Updated in 5a1430e422fc42dcea0bbb1a6643871d4147f43b